### PR TITLE
added implementation for seq property path

### PR DESCRIPTION
**Pull request by patofp**
# Description

This PR adds support for `seq` property path. E.g:

```
PREFIX foaf: <http://xmlns.org/foaf/0.1/>

SELECT ?o
WHERE {
 ?s foaf:knows/foaf:knows/foaf:name ?o .
}
```

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

### CODE REVIEW

File: modules/engine/src/main/scala/com/gsk/kg/engine/Engine.scala [hunk 1]
```diff
@@ -237,7 +257,8 @@ object Engine {
       Foldable[ChunkedList].fold(
         quads.mapChunks { chunk =>
           val condition = composedConditionFromChunk(df, chunk)
-          applyChunkToDf(chunk, condition, df)
+          val a         = applyChunkToDf(chunk, condition, df)
+          a
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/Engine.scala:261]: Nitpick: can we leave as it was? hehehe

File: modules/engine/src/main/scala/com/gsk/kg/engine/PropertyExpressionF.scala [hunk 2]
```diff
@@ -15,35 +16,41 @@ import com.gsk.kg.sparqlparser.Result
 
 object PropertyExpressionF {
 
+  type ColOrDf = Either[Column, DataFrame]
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/PropertyExpressionF.scala:19]: 👍🏻

**Review by pepegar — APPROVED** [review]
great job mate!

File: modules/engine/src/main/scala/com/gsk/kg/engine/Engine.scala (hunk 1 above)
Comment (patofp) [reply modules/engine/src/main/scala/com/gsk/kg/engine/Engine.scala]: uuups! it was just for debug purpose 😆

**Pull request merged**